Add API Gateway Module documentation (#493)
kyma-project/api-gateway: docs/contributor/04-10-technical-design.md

#### Reconciliation
The [APIGateway CR](../user/03-technical-reference/custom-resources/apirule/01-40-apirule-custom-resource.md) is reconciled with each change. If no changes have been made, the default reconciliation interval 
of 10 hours set by the [Kubernetes controller-runtime](https://pkg.go.dev/sigs.k8s.io/controller-runtime) is used.  
In the event of a failure during the reconciliation process, the default [Kubernetes controller-runtime](https://pkg.go.dev/sigs.k8s.io/controller-runtime) exponential backoff requeue behavior is used.

### API Rule Controller

The API Rule Controller is a [Kubernetes controller](https://kubernetes.io/docs/concepts/architecture/controller/), which is implemented with the [Kubebuilder](https://book.kubebuilder.io/) framework. 
The controller is responsible for handling the [APIRule CR](../user/03-technical-reference/custom-resources/apirule/01-40-apirule-custom-resource.md).  
Additionally, the controller watches the [api-gateway-config](../user/03-technical-reference/custom-resources/apirule/01-40-apirule-custom-resource.md#jwt-access-strategy) to configure the JWT handler.

#### Reconciliation

> The [APIRule CR](../user/03-technical-reference/custom-resources/apirule/01-40-apirule-custom-resource.md) is reconciled with each change. If no changes have been made, then the default reconciliation interval of one hour is used. 

Reviewer: ```suggestion
The [APIRule CR](../user/03-technical-reference/custom-resources/apirule/01-40-apirule-custom-resource.md) is reconciled with each change. If no changes have been made, process occurs at the default interval of 10 hours.
```

> This interval can be adjusted with the [API Gateway Operator paramteres](../user/03-technical-reference/configuration-parameters/01-10-api-gateway-operator-parameters.md).  

Reviewer: ```suggestion
You can use the [API Gateway Operator paramteres](../user/03-technical-reference/configuration-parameters/01-10-api-gateway-operator-parameters.md) to adjust this interval.  
```

> In the event of a failure during the reconciliation, the controller will perform the reconciliation again after 1 minute.

Reviewer: ```suggestion
In the event of a failure during the reconciliation, the controller performs the reconciliation again after one minute.
```

> The following diagram illustrates the process of APIRule reconciliation and the resources created:

Reviewer: ```suggestion
The following diagram illustrates the reconciliation process of APIRule and the created resources:
```
